crates/common/mqtt_channel/src/tests.rs

```diff
@@ -500,4 +500,43 @@ mod tests {
 
         Ok(())
     }
+
+    #[tokio::test]
+    #[serial]
+    async fn ensure_that_last_will_message_is_delivered() -> Result<(), anyhow::Error> {
+        let broker = mqtt_tests::test_mqtt_broker();
+        let topic = "lastwilltopic";
+        let mut messages = broker.messages_published_on(topic).await;
+
+        // An mqtt process publishing messages
+        // must ensure the messages have been sent before process exit.
+        std::thread::spawn(move || {
+            tokio::runtime::Builder::new_current_thread()
+                .enable_all()
+                .build()
+                .unwrap()
+                .block_on(async {
```

Comment:
I don't understand what is done and why.

Reply:
Here I create a `mqtt client` with `lastwill` message and publish a message and disconnect it.
On the subscriber (in assert) waiting for all the messages, including the `lastwll` message.
Is there something wrong with the test?.
I think the `rumqttd` broker is not working for the `lastwill` message.

Reply:
What I don't understand is the way you spawn this task. Why a thread not a regular tokio task?

Reply:
Yes, now using the `tokio task`.